### DISCUSSION TITLE
Branch Predictor

### DISCUSSION THREAD

**Discussion by SteliosKaragiorgis** [ChampSim/ChampSim#472] [Q&A]
Hello. Im trying to implement my branch predictor. As i relised i need to implement these 3 methods:

`void O3_CPU::initialize_branch_predictor() {} `
`uint8_t O3_CPU::predict_branch(uint64_t ip){}`
`void O3_CPU::last_branch_result(uint64_t ip, uint64_t branch_target, uint8_t taken, uint8_t branch_type){}`

So as I relised I take only instruction's ip, branch target, taken and branch type.
So my question is: is it a way to access more information about the instruction as source registers, destination registers,registers_instrs_depend_on_me, num_reg_dependent, etc..?

Of course I want to access these information without change anything on champsim code, I want to access these information from my predictor

Thanks in advance

**Comment by ngober** [comment]
As discussed in #451, there is currently no interface to these values. ChampSim models a very early branch prediction, when only the instruction pointer is known. Any further information would require a modification to the module interface.

**Reply by SteliosKaragiorgis** [reply]
Thank you!!
